### DISCUSSION TITLE
Does Anyone Know How to do Networking Effects with Combine in the TCA architecture ?

### DISCUSSION THREAD

**Discussion by Dominic263** [pointfreeco/swift-composable-architecture#1233] [Q&A]
Hi folks, 

I am having a challenge when it comes to doing any networking using combine for my mock TCA app. I decided to learn TCA by actually building a project a few weeks ago but I am having issues with network effects. I am trying to fetch data from https://catfact.ninja/facts to display cat facts in the app, and I have somehow realized that there is an error happening because the replaceError(with: ) function is being invoked every time I fetch data. 
This is how I modeled my Data as per the schema from the site I am trying to execute the get request. 
![Screen Shot 2022-08-01 at 5 30 41 PM](https://user-images.githubusercontent.com/59946287/182249964-c440601e-4d74-4135-b908-9fa7f8af4edc.png)
 and this is the snippet of code from my reducer which shows how I try to fetch data using URLSession 
![Screen Shot 2022-08-01 at 5 32 56 PM](https://user-images.githubusercontent.com/59946287/182250243-dc8c2bb7-bbf5-4a72-b661-40544d50edbb.png)

Is there something I am doing wrong here and if so how can I correct that? Thanks for your help and let me know if I should provide more information.

**Comment by stephencelis** [comment]
@Dominic263 Looking at the JSON payload, I think the issue is you are trying to decode `[CatFact].self` from a response that nests that array (`{"data":[...],...})`. That means you need an additional wrapper type:

```swift
struct CatFactResponse: Decodable {
  var data: [CatFact]
}
```

Which you can use when you call `.decode` on the Combine publisher.

A couple notes on debugging tips:

If you call `.print()` on the publisher _before_ `.replaceError`, you should get info about the error in the console, which will probably be a decoding error:

```swift
.print()
.replaceError(with: [...])
```

**Comment by Dominic263** [comment]
works like magic. Thanks, Stephen!

![Screen Shot 2022-08-01 at 6 57 21 PM](https://user-images.githubusercontent.com/59946287/182259224-8822aa4f-6e48-46b7-9c2e-307d7fe98802.png)

**Reply by Dominic263** [reply]
with regards to debugging, I am also a newbie in Xcode and I tried to insert the print statement before replacerError but it doesn't seem to work. Nothing prints out in the Debug area. I tried as well to go to View > DebugArea > Activate Console, then built and ran my project but when I execute fetch by tapping the stats button there are no print statements shown. Am  missing something?

**Reply by stephencelis** [reply]
@Dominic263 I think we'd need to see the exact publisher code and more screenshots to debug further, sorry!
